### ISSUE TITLE
Slicer dashboard tests are failing

### ISSUE THREAD

**Issue by fedorov** [QIICR/dcmqi#137]
15 tests fail on windows (there is no log, so not sure what is going on)

http://slicer.cdash.org/viewTest.php?onlyfailed&buildid=925024

On linux and mac, 2 parametric map tests are failing, need to look at the temp directory initialization:

http://slicer.cdash.org/testDetails.php?test=7703552&build=924818

**Comment by fedorov** [comment]
Some tests are still failing - it may be a separate issue, but I will keep this issue open until those are resolved.

![image](https://cloud.githubusercontent.com/assets/313942/20400445/b5804fbe-acc2-11e6-8d2d-e15f82d0acf1.png)

**Comment by jcfr** [comment]
On windows, running test locally gives the following output:

```
C:\D\N\S-1-E-b\DCMQI-build\dcmqi-build>C:\D\Support\CMake-3.5.0-rc3\bin\ctest.exe -C Release
Test project C:/D/N/S-1-E-b/DCMQI-build/dcmqi-build
      Start  1: itkimage2paramap_hello
 1/17 Test  #1: itkimage2paramap_hello .....................   Passed    1.31 sec
      Start  2: itkimage2paramap_makeParametricMap
 2/17 Test  #2: itkimage2paramap_makeParametricMap .........   Passed    0.81 sec
      Start  3: itkimage2paramap_makeParametricMapFP
 3/17 Test  #3: itkimage2paramap_makeParametricMapFP .......   Passed    0.31 sec
      Start  4: paramap2itkimage_hello
 4/17 Test  #4: paramap2itkimage_hello .....................   Passed    0.12 sec
      Start  5: paramap2itkimage_makeNRRDParametricMap
 5/17 Test  #5: paramap2itkimage_makeNRRDParametricMap .....   Passed    0.90 sec
      Start  6: paramap2itkimage_makeNRRDParametricMapFP
 6/17 Test  #6: paramap2itkimage_makeNRRDParametricMapFP ...   Passed    0.48 sec
      Start  7: itkimage2segimage_hello
 7/17 Test  #7: itkimage2segimage_hello ....................   Passed    0.16 sec
      Start  8: itkimage2segimage_makeSEG
 8/17 Test  #8: itkimage2segimage_makeSEG ..................   Passed    0.27 sec
      Start  9: segimage2itkimage_hello
 9/17 Test  #9: segimage2itkimage_hello ....................   Passed    0.12 sec
      Start 10: segimage2itkimage_makeNRRD
10/17 Test #10: segimage2itkimage_makeNRRD .................   Passed    0.27 sec
      Start 11: seg_meta_roundtrip
11/17 Test #11: seg_meta_roundtrip .........................   Passed    0.73 sec
      Start 12: tid1500writer_hello
12/17 Test #12: tid1500writer_hello ........................   Passed    0.27 sec
      Start 13: tid1500writer_example
13/17 Test #13: tid1500writer_example ......................   Passed    0.27 sec
      Start 14: tid1500writer_ct-liver
14/17 Test #14: tid1500writer_ct-liver .....................   Passed    0.20 sec
      Start 15: tid1500reader_hello
15/17 Test #15: tid1500reader_hello ........................   Passed    0.14 sec
      Start 16: tid1500reader_ct-liver
16/17 Test #16: tid1500reader_ct-liver .....................   Passed    0.17 sec
      Start 17: tid1500_meta_roundtrip
17/17 Test #17: tid1500_meta_roundtrip .....................   Passed    0.09 sec

100% tests passed, 0 tests failed out of 17

Label Time Summary:
itkimage2paramap    =   2.43 sec (3 tests)
paramap2itkimage    =   1.51 sec (3 tests)
segmentation        =   1.54 sec (5 tests)
tid1500             =   1.05 sec (5 tests)
tid1500reader       =   0.09 sec (1 test)
```

That said, running in parallel make some of the test fail. 

```
C:\D\N\S-1-E-b\DCMQI-build\dcmqi-build>C:\D\Support\CMake-3.5.0-rc3\bin\ctest.exe -C Release -j4
Test project C:/D/N/S-1-E-b/DCMQI-build/dcmqi-build
      Start  5: paramap2itkimage_makeNRRDParametricMap
      Start  6: paramap2itkimage_makeNRRDParametricMapFP
      Start  1: itkimage2paramap_hello
      Start  2: itkimage2paramap_makeParametricMap
 1/17 Test  #2: itkimage2paramap_makeParametricMap .........   Passed    0.20 sec
      Start 11: seg_meta_roundtrip
 2/17 Test  #1: itkimage2paramap_hello .....................   Passed    0.45 sec
 3/17 Test #11: seg_meta_roundtrip .........................   Passed    0.11 sec
 4/17 Test  #6: paramap2itkimage_makeNRRDParametricMapFP ...***Failed    0.56 sec
      Start  8: itkimage2segimage_makeSEG
      Start 13: tid1500writer_example
      Start 14: tid1500writer_ct-liver
 5/17 Test #13: tid1500writer_example ......................   Passed    0.16 sec
 6/17 Test  #5: paramap2itkimage_makeNRRDParametricMap .....   Passed    0.92 sec
 7/17 Test #14: tid1500writer_ct-liver .....................   Passed    0.17 sec
      Start 10: segimage2itkimage_makeNRRD
      Start  3: itkimage2paramap_makeParametricMapFP
      Start 12: tid1500writer_hello
 8/17 Test  #8: itkimage2segimage_makeSEG ..................   Passed    0.41 sec
 9/17 Test #10: segimage2itkimage_makeNRRD .................   Passed    0.19 sec
10/17 Test #12: tid1500writer_hello ........................   Passed    0.17 sec
      Start 16: tid1500reader_ct-liver
      Start 17: tid1500_meta_roundtrip
      Start  7: itkimage2segimage_hello
11/17 Test  #3: itkimage2paramap_makeParametricMapFP .......   Passed    0.28 sec
12/17 Test #16: tid1500reader_ct-liver .....................   Passed    0.14 sec
13/17 Test #17: tid1500_meta_roundtrip .....................   Passed    0.12 sec
      Start 15: tid1500reader_hello
      Start  4: paramap2itkimage_hello
      Start  9: segimage2itkimage_hello
14/17 Test  #4: paramap2itkimage_hello .....................   Passed    0.10 sec
15/17 Test #15: tid1500reader_hello ........................   Passed    0.11 sec
16/17 Test  #7: itkimage2segimage_hello ....................   Passed    0.31 sec
17/17 Test  #9: segimage2itkimage_hello ....................   Passed    0.10 sec

94% tests passed, 1 tests failed out of 17

Label Time Summary:
itkimage2paramap    =   0.94 sec (3 tests)
paramap2itkimage    =   1.58 sec (3 tests)
segmentation        =   1.12 sec (5 tests)
tid1500             =   0.75 sec (5 tests)
tid1500reader       =   0.12 sec (1 test)

Total Test time (real) =   1.75 sec

The following tests FAILED:
          6 - paramap2itkimage_makeNRRDParametricMapFP (Failed)
Errors while running CTest
```

I will look into re-runing the test in parallel after removing the intermediate output. I suspect some of the dependency are not properly set ...

**Comment by jcfr** [comment]
PR #147 addresses the failure of paramap2itkimage_makeNRRDParametricMap tests

**Comment by fedorov** [comment]
tests are still failing, but @che85 I would hope this is not critical for RSNA2016 milestone!

http://slicer.cdash.org/viewTest.php?onlyfailed&buildid=926440

**Comment by che85** [comment]
I have no idea, why that is happening. I can only imagine, that the output path for the converters is messed up somehow...

**Comment by jcfr** [comment]
This issue should now be resolved.

Parallel execution of test was failing because dependencies were not properly set for two reasona:
1. an issue with `dcmqi_add_test`
2. dependencies not specified in paramaps tests

**Comment by fedorov** [comment]
Considering how difficult it is to debug Slicer dashboard issues, and that dcmqi CI dashboards provide more robust testing mechanism, I close this issue.
